src/stratigraphy/benchmark/score.py

```diff
@@ -193,13 +215,99 @@ def evaluate_metadata(predictions: dict) -> tuple[dict, pd.DataFrame]:
     except ZeroDivisionError:
         coordinate_recall = 0
 
+    groundwater_tp = 0  # correct prediction
+    groundwater_fn = 0  # no predictions, i.e. None
+    groundwater_fp = 0  # wrong prediction
+
+    # there is no such thing as true negatives as each document has coordinates - though not necessarily specified
+    number_groundwater_predictions = 0
+    for file_name, file_prediction in predictions.items():
+        if file_prediction.metadata_is_correct["groundwater_information"]:
+            groundwater_tp += 1
+            number_groundwater_predictions += 1
+            document_level_metrics_groundwater_information["document_name"].append(file_name)
+            document_level_metrics_groundwater_information["groundwater_information"].append(1)
+        elif file_prediction.metadata_is_correct["groundwater_information"] is None:
+            groundwater_fn += 1
+            number_groundwater_predictions += 1
+        else:
+            groundwater_fp += 1
+            number_groundwater_predictions += 1
+            document_level_metrics_groundwater_information["document_name"].append(file_name)
+            document_level_metrics_groundwater_information["groundwater_information"].append(0)
+
+    try:
+        groundwater_precision = groundwater_tp / (groundwater_tp + groundwater_fp)
+    except ZeroDivisionError:
+        groundwater_precision = 0
+    try:
+        groundwater_recall = groundwater_tp / (groundwater_tp + groundwater_fn)
+    except ZeroDivisionError:
+        groundwater_recall = 0
```

Comment:
This code is now more or less duplicated three times in this method (coordinates, ground water, ground water depth). Would it be possible to create a new method that computers the metrics for any of these properties (taking the exact property as a parameter) and that can then be called three times?